src/canvas-extractor/tests/graphql/extractor/test_extractor_integration.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import os
+import pytest
+
+from pandas import DataFrame
+from pathlib import Path
+from sqlalchemy import create_engine
+
+from edfi_canvas_extractor.graphql.extractor import GraphQLExtractor
+from edfi_canvas_extractor.graphql.schema import query_builder
+
+from edfi_canvas_extractor.graphql.assignments import assignments_synced_as_df
+from edfi_canvas_extractor.graphql.courses import courses_synced_as_df
+from edfi_canvas_extractor.graphql.enrollments import enrollments_synced_as_df
+from edfi_canvas_extractor.graphql.sections import sections_synced_as_df
+from edfi_canvas_extractor.graphql.students import students_synced_as_df
+
+
+@pytest.fixture(scope="class")
+def gql_class():
+    CANVAS_BASE_URL = os.environ['CANVAS_BASE_URL']
+    CANVAS_ACCESS_TOKEN = os.environ['CANVAS_ACCESS_TOKEN']
+    START_DATE = "2021-01-01"
+    END_DATE = "2030-01-01"
+
+    gql = GraphQLExtractor(
+        CANVAS_BASE_URL,
+        CANVAS_ACCESS_TOKEN,
+        "1",
+        START_DATE,
+        END_DATE,
+        )
+    yield gql
+
+
+@pytest.fixture(scope="class")
+def graphql_class(gql_class):
+    query = query_builder(account_id=1)
```

Comment:
SHOULD: Instead of building the query here and then calling `get_from_canvas`, we should use use `gql_class.run()` which performs the same operations and also sets `has_data` flag which can then be asserted in the test.